src/internal/m365/collection/exchange/collection.go

```diff
@@ -234,6 +238,10 @@ func (col *prefetchCollection) streamItems(
 
 			stream <- data.NewDeletedItem(id)
 
+			if col.Counter.IncRead(count.StreamItemsRemoved)%1000 == 0 {
+				logger.Ctx(ctx).Infow("item stream progress", "stats", col.Counter.Values())
```

Comment:
Do we need to log all `counter.Values()` everywhere? Can we just log the count key which is mutating?

I see that the `defer` block is logging all values already, that should be sufficient.

Reply:
Is there any drawback to logging the full set?

Reply:
Seems unnecessary, especially in this case. In a pathological case you'd have 100s of stats, all the same except for one which is relevant & mutating. Harder to inspect while reviewing logs, it's like finding needle in a haystack. Unless you know the key name you are looking for.  Also, more chatty logs = more scroll time. My fingers hurt!

Don't know if there is a huge penalty for serializing the map to logger every time. But given this is a concurrent map, there might be some? Still, an extreme case and I'm not raising this concern.

Reply:
Hmm. Beyond this particular case, my concern is that we will end up using `.Values()` pattern everywhere. If we are doing say an `Infow` log with a key, val pair, we shouldn't dump all the stats by default unless they are meaningful in that context.